ast/compile.go

```diff
@@ -928,6 +969,29 @@ func parseSchema(schema interface{}) (types.Type, error) {
 		return orType, nil
 	}
 
+	if subSchema.AllOf != nil {
+		subSchemaArray := subSchema.AllOf
+		allOfResult, err := mergeSchemas(subSchemaArray)
+		if err != nil {
+			return nil, err
+		}
+
+		if subSchema.Types.IsTyped() {
+			if (subSchema.Types.Contains("object") && allOfResult.Types.Contains("object")) || (subSchema.Types.Contains("array") && allOfResult.Types.Contains("array")) {
+				var temp []*gojsonschema.SubSchema
+				temp = append(temp, allOfResult, subSchema)
+				objectOrArrayResult, err := mergeSchemas(temp)
```

Comment:
I might be missing something but I think that's the same as
```suggestion
				var temp []*gojsonschema.SubSchema
				temp = append(temp, allOfResult, subSchema)
				objectOrArrayResult, err := mergeSchemas(append(allOfResult, subSchema))
```
or something close 😄 

Maybe it would also read nicely if we changed `mergeSchemas` to
```go
func mergeSchemas(schemas ...*gojsonschema.SubSchema) (*gojsonschema.SubSchema, error) {
```